js/ui/file_manager/ui.file_manager.editing.js

```diff
@@ -160,43 +213,188 @@ class FileManagerEditingControl extends Widget {
+            if(!context.singleRequest) {
+                this._notificationControl.completeOperationItem(operationInfo, info.index, context.commonProgress);
+            }
+        }
+    }
+
+    _handleActionError(operationInfo, context, errorInfo, singleRequest) {
+        singleRequest = ensureDefined(singleRequest, context.singleRequest);
+
+        operationInfo.hasError = true;
+
+        if(singleRequest) {
+            this._handleSingleRequestActionError(operationInfo, context, errorInfo);
+        } else {
+            this._handleMultipleRequestActionError(operationInfo, context, errorInfo);
+        }
+    }
+
+    _handleSingleRequestActionError(operationInfo, context, errorInfo) {
+        const itemInfo = context.getItemForSingleRequestError();
+        const errorText = this._getErrorText(errorInfo, itemInfo);
+
+        context.processSingleRequestError(errorText);
+        const operationErrorInfo = this._getOperationErrorInfo(context);
+        this._notificationControl.completeSingleOperationWithError(operationInfo, operationErrorInfo);
+
+        if(context.multipleItems) {
+            this._raiseOnSuccess(context.onlyFiles);
+        }
     }
 
-    _onFileProviderError(errorInfo, fileItems) {
-        const fileItem = fileItems[errorInfo.index];
-        this._raiseOnError(errorInfo.errorId, fileItem);
+    _handleMultipleRequestActionError(operationInfo, context, errorInfo) {
+        const itemInfo = context.getItemForMultipleRequestError(errorInfo.index);
+        const errorText = this._getErrorText(errorInfo, itemInfo);
+
+        context.processMultipleRequestError(errorInfo.index, errorText);
+        const operationErrorInfo = this._getOperationErrorInfo(context);
+        this._notificationControl.addOperationDetailsError(operationInfo, operationErrorInfo);
+    }
+
+    _getOperationErrorInfo(context) {
+        const detailError = context.errorState.currentDetailError;
+        return {
+            commonErrorText: context.errorState.commonErrorText,
+            item: detailError.itemInfo ? this._getItemProgressDisplayInfo(detailError.itemInfo) : null,
+            itemIndex: detailError.itemIndex,
+            detailErrorText: detailError.errorText
+        };
+    }
+
+    _getErrorText(errorInfo, itemInfo) {
+        const itemName = itemInfo ? itemInfo.fileItem.name : null;
+        return FileManagerMessages.get(errorInfo.errorId, itemName);
+    }
+
+    _getItemProgressDisplayInfo(itemInfo) {
+        return {
+            commonText: itemInfo.fileItem.name,
+            imageUrl: this._getItemThumbnail(itemInfo)
+        };
     }
 
     _tryUpload(destinationFolder) {
-        this._uploadFolder = destinationFolder && destinationFolder[0] || this._getCurrentDirectory();
+        this._uploadDirectoryInfo = destinationFolder && destinationFolder[0] || this._getCurrentDirectory();
         this._fileUploader.tryUpload();
     }
 
+    _getItemInfosForUploaderFiles(files, parentDirectoryInfo) {
```

Comment:
I think, that best home for the method is controller class. Main concept of controller class was described above.

Reply:
Agree. Fixed.